src/widget/woverview.h

```diff
@@ -78,7 +78,11 @@ class WOverview : public WWidget, public TrackDropTarget {
     }
 
     double getTrackSamples() const {
-        return m_trackSamplesControl->get();
+        if (m_trackLoaded) {
+            return m_trackSamplesControl->get();
+        } else {
+            return 0.0;
+        }
     }
```

Comment:
IMO this only fixes the symptom of a more complex issue. If there is no track loaded, the CO should already be `0.0`...

Reply:
Do you have a better solution? 
The complex issue is that the track change is propagated via the Qt signal queue while the track samples control is changed instantly as an atomic. 
m_trackLoaded bool ensures that the value is not used during the transition.

Reply:
I see, so its basically a data-race.
The problem I still have with this solution is that the waveform assumes that CO values `<= 0.0` mean that no track is loaded. Or rather, it relies on the fact that that is always the case (which it isn't due to the data-race). I think it would be better to move the `m_trackLoaded != nullptr` check to callers where they should actually check that.

Reply:
The track length is set here to an invalid value from the worker thread: 
https://github.com/mixxxdj/mixxx/blob/ceaedb8443714459bc3544faab957d6903eba7b3/src/engine/enginebuffer.cpp#L526
in a responds from a new track loaded into the player. 

The Waveform overview visualizes the track loading process. It has already the new track before the engine had evicted the old one. In this situation is guarded by the m_trackLoaded boolean. 

I will add a comment.

Reply:
I think a outdated waveform update signal cause a paint event with these inconstant data.